BDArmory/Weapons/Missiles/MissileBase.cs

```diff
@@ -665,6 +679,13 @@ protected void UpdateLaserTarget()
                 }
                 else //lost active laser target, home on last known position
                 {
+                    if (!hasIOG) {
+                        _LaserFailTime += Time.fixedDeltaTime;
+                        if (maxLaserFailTime < _LaserFailTime) {
+                            guidanceActive = false;
+                            Detonate();
+                        }
+                    }
```

Comment:
So laser missiles now... detonate slightly earlier if they lose laser lock? 
Laser missiles aren't launched unless a laserdot is present (or, I guess, you make a missile with laser terminal guidance), so you can't really loft them ballistically until they enter active range.
Reworking missileFire to get the AI to use 'unguided' missiles has been a project of mine on the backburner, which would permit something like this perhaps, but at present this just going to terminate LASMs that have lost lock, assuming they haven't already slammed into the ground.

Reply:
the ideia is something like early Laser guided weapons , if in a situation on I.A vs I.A if the defending one use smoke and the attacking can't change or lock other thing the missile just detonate itself because of the lack of guidance